support/js/support_system_file.js

```diff
@@ -58,6 +58,38 @@ function support_system_file_readLine (file_ptr) {
   return line
 }
 
+function support_system_file_readChars (length, file_ptr) {
+  const readBuf = Buffer.alloc(length)
+  const bytesRead = support_system_file_fs.readSync(file_ptr.fd, readBuf, 0, length, null)
+  if (bytesRead === 0) {
+    file_ptr.eof = true
+  }
+  const chars = file_ptr.buffer.slice(0, bytesRead + 1).toString('utf-8')
+  return chars
+}
+
+function support_system_file_readChar (file_ptr) {
+  const readBuf = Buffer.alloc(1);
+  if (support_system_file_fs.readSync(file_ptr.fd, readBuf, 0, 1) === 0) {
+    file_ptr.eof = true
+    // No bytes read, getChar from libc returns -1 in this case.
+    return String.fromCharCode(-1)
+  } else {
+    return readBuf.toString('utf-8')
+  }
+}
+
+function support_system_file_flush (file_ptr) {
+  try{
+    support_system_file_fs.fdatasyncSync(file_ptr.fd)
+    return 0
+  }catch(e){
+    process.__lasterr = e
+    file_ptr.eof = true
+    return String.fromCharCode(-1)
```

Comment:
This looks like it's returning a number (0)  in one case and a string in the other case.

Reply:
adressed